snapcraft.yaml

```diff
@@ -1605,7 +1521,9 @@ parts:
       - libudev1
       - libunity-dev
       - libuuid1
+      - libva-dev
       - libvorbisfile3
+      - libwayland-dev
```

Comment:
In "Rely on gpu-2404" you remove the libraries corresponding to these packages. Is "gpu-2404" using these packages? Because if it isn't, then maybe we should rely here on the headers from gpu-2404...

Reply:
If you end up needing a newer version of libwayland than is in the gpu snap (which will, by default, be the 24.04 version), it's either the app or the GNOME snap again that will need to bring it in.

The GPU snap does not have (and shouldn't have) headers, it's not meant to be developed against, because you get the assurance that the libraries in there are 24.04 ABI-compatible. That's different from the GNOME stack because users may swap the gpu snap for another (newer version, proprietary stack etc.), so the contract has to be on the ABI level.

Reply:
Oh, ok... you are right.

Reply:
But as in the nature of development in the gnome world, they are gonna update the needed version of `wayland` and I really think it's always better to have latest version of both `wayland` and `libva`. If we eventually get to a point where we'll need to bring back the wayland, libva and other libraries, what is the use of removing those. Please kindly rethink about this.

Reply:
libwayland is the less problematic one. Personally I'd keep to the archive as long as that's possible. More, I'd switch the sources for all those to the latest Ubuntu stable one so that we get our security team's advantage.

But libva we should stick to 24.04 versions. Otherwise no one can build a GPU enablement snaps that has VAAPI support that GNOME snaps would work with. The immediate example of that is Apple Silicon - for 22.04 that meant maintaining a separate track for the GNOME stack. But then there's hardware enablement, ARM stacks... we really should strive to have broad buy-in into a solution that doesn't require everyone to branch out for every hardware stack out there.